### DISCUSSION TITLE
A bunch of little ideas and one big one

### DISCUSSION THREAD

**Discussion by ChuckStarchaser** [endless-sky/endless-sky#7514] [Ideas]
First I want to commend the team.  I began playing this game with version 0.8.9, and when yesterday I began playing with version 9.16.1 it was night and day, a breath of fresh air.  Awesome progress!  Little things matter much sometimes, such as the slower scrolling of the star background, which gives the newer version so much more depth.

But on with this post.

COSMETIC IDEAS FIRST (NO PARTICULAR ORDER):

* C1:  Asteroids improvements:  Similar to the idea already implemented with the slower moving background, asteroids could come, for example, in two sizes:  Big ones that move slowly and are always behind any ships (further in the depth of field), and smaller ones that move faster (but not as fast as they do presently;  that's too fast), and that can be mined.  Optionally, the density of the smaller asteroids could vary from almost none where there are no larger asteroids in the background, for example, to a density almost as large as presently around larger asteroids.  Optionally, the smaller asteroids can move as if they were orbiting the larger ones.  Optionally, there could be some small asteroids that orbit larger ones, and occasionally free small asteroids.  The orbiting ones could be mine-able as they orbit above the plane but become darker, less visible, as they orbit further, below the larger ones.

* C2:  Interstellar flight without hyperdrive:  It could be fun if it were possible, even if almost impossible.  Suppose you fly your ship in the direction of a nearby star but don't press J or don't have hyperdrive or run out of fuel and don't have a scoop:  With this change, it would be possible to eventually arrive to that other star, after flying for a very long time, let's just say 45 minutes of player time.  Why?  Mostly just for intellectually cosmetic reasons:  Adding a feeling of continuity to the map.  It convinces you more that it is real if you CAN fly it without a jump-drive.  Also, having a more immersive way of recovering from bad situation without having to resort to the most anti-immersive action in any game:  having to reload.  Also, allowing for game possibilities, such as some secret place, a secret wormhole, for example, to exist but not be known until someone in the story tells you "in system such and such, fly West young man", and you go there and fly West for like a full minute straight and, voila!, find a secret base or jump or whatever.  Also because if the full map represents the galaxy, the distance between most stars is in the order of 1000 light years or more;  which means that there are many other stars in-between that are not shown in the map simply because there are no jump lines connecting them, right?

* C3:  Slowing down space messages so that they can actually be read.  Okay, perhaps I'm a slow reader, but whenever there are messages coming up the bottom left of the screen, I sometimes try to read them, but they pass too fast for me;  I can never read the full line before it disappears.  Also, there are too many messages coming through, with the likely result that actually NOBODY reads them, I suspect.  If a message came once in a blue moon, they would feel more valuable or precious, even if just decorative;  and then one would perhaps actually read messages that are important coming from there, which I was surprised to discover, such as planetary communications.

*  C4:  Images of faces of characters who give you missions.  It is very painful not to have the slightest idea what these people you work so much with, like during the Free Worlds campaign for example, look like.  I know, this is a lot of work;  but all I want to say is it should have a high priority.  One way to ameliorate the situation is to add just one rule for story-makers:  Make it mandatory for them to submit their stories WITH pictures for the characters in it.  Writers of stories already have an image in their heads of what their characters look like, so it should not be too difficult for them to find images that resemble the people they imagine.
  In the case of alien races, this needs a lot more work, of course.  You could go to the blenderartists forum and say "Ahem, we could use some help, if anyone is willing;  we need a few models of aliens",  and there would be takers, I guarantee it.  Myself, I'm good at modeling ships, machines, stations, structures;  I'm not good at modeling living things.

*  C5:  Text reduction.  I don't mean less story, of course;  I mean less text as interface, more image based interface.  The question about any text in the game should be "is it poetic?"  If not, replace it.  Descriptions of weapons, if captivating, are good.  Conversations are good.  Ancient books in an alien library are good.  But NOT text or numbers for management of game assets.  That's a no-no.
  For the smallest example:  The number of a particular type of outfit installed in the ship could be represented by repeating the image or icon representing it, instead of by it being shown as a number.
  This is KEY to having a game that feels real, in the immersiveness sense:  The interface should try not to mix domains.  Text and numbers for managing things is in the realm of business, work, and everything you play a game to forget and get away from, not to face yet again.  This was the problem with Masters of Orion 3.  I had been a big fan of MOO1, and an even bigger fan of MOO2.  When the company decided to go for a MOO3 I was very excited, like most fans, so I joined their development forum, and I downloaded samples of the work in progress, but I noticed something was wrong:  They had replaced many of the pretty looking image based management things with numbers and text and scroll-down menus and check-boxes.  And I told them clearly what I thought of it, namely that they were KILLING THE FUN by getting rid of the graphics, making the game as fun as working with DBase2 or Excell or some other work application.  The whole game was like a stardard windows application except for a few nice images in frames...  What?  I told them the game should not be graphical ONLY when you are fighting a battle in space;  it should be graphical everywhere, like MOO2 was.  The team lead got so angry at me that he banned me from posting, but all the other people continued posting telling him that I was right ... Then he locked the thread to shut them up.  Anyways, MOO3 finally released and it was a flop;  nobody liked it.  Total disaster.
  Imagine this was an RPG:  Most RPG's don't switch from graphical mode to a standard windows interface and business fonts when you enter the weapons shop, or start a conversation with an NPC;  they give you continuity of the medieval style experience and feel by making every interface, every mode, every game state, graphics based, all using common decore and style.  Even the font used for text is a custom font meant to look gothic, and like it's ink on parchement.
  Of course this is in the future, not in the past;  but same difference.
  Another solution is what Privateer (WC) did:  You see text and numbers for management of some things,  but it appears on the (old style) LCD display of a personal tablet computer that you had in-game and almost filled the screen when you pressed C.  And you could press the buttons on it on-screen to control it.  In other words, you CAN have a text and numbers management interface AND remain immersive ***IF*** it is shown on a display of a device IN GAME, as opposed to appearing right on your own computer screen.  I'd call this "a solution by framing".
  I was suggesting elsewhere in this forum that management of the fleet, and of how cargo is loaded onto the fleet, could use a graphical interface.
  Missions could also use a graphical interface, if we had pictures of the characters.
  And same thing with Trading.  Plastics:  like bottles for recycling.  Food:  a cornucopia.  Metal:  Ingots.  Etc.  This could be on a screen in-game provided for traders at the trade centers, since most of the actual dealing with cargo would be done by robots.

*  C6:  Larger systems, with more stuff in them.  I have a belief that a perfect game is one where about half the time is spent in action and half the time is spent reading, planning, managing...  I have no argument to offer;  I just find that most games that I found really rewarding and immersive were about 50-50 in this sense.  Take a game like Privateer:  About half the time (player time) flying, and the other time going places at a base, buying and selling, selecting missions, upgrading the ship, etc.  Same thing with Diablo.  Right now ES is about 40-60 or 30-70:  You fly rapidly, then spend longer at the base.  The flying time should go up, in my opinion.  And some ways to extend flight time is to make it not possible to engage hyperdrive until you are "far enough away from strong gravitational fields and dangerous obstacles" ... In other words, you have to fly away from the planets, and past the asteroid belt, and to have no enemies near, THEN you can warp.

*  C7:  Add image to player character creation:  Choices of stock images, plus ability to pick your own photo or artistic creation.

FUNCTIONAL IDEAS (NO PARTICULAR ORDER):

*  F1:  Making the generator algorithm make more sense relative to geography, such as having pirate generation that is maximal at pirate systems and tapers away from them,  and that configures itself automatically by reading the map, instead of manually plugging magic numbers into every system file, which is what I suspect is the present method.  Optionally, the generator could have a 2D "Life"-like simulation engine running that generates continuous faction density waves radiating from all systems in the map, and then occasionally collapses the probability wave to generate a group of NPC's  ;-)

*  F2:  Making ship design a far more complex proposition.  Right now the dynamic is that the player needs AND wants EVERY outfit installed in every ship.  This results in that size is all that matters.  The more outfit space a ship has, the more it will kick.  The ONLY question is whether the shields should be from the Hai,  should the batteries be Coalition?, etceteras.
  This is too simpleton-ic and boring.
  One outstandingly effective way to break this monotony is to have a much greater variety of types of outfits than there is space in any ship for.  For example, a list of outfits for improving the effectiveness of your beam weapons could include,
1)  Range extender
2)  Ultra-low vibration weapon mounts (double the accuracy of any beam weapon)
3)  Target flight path predictor for aiming lead (velocity only)
4)  Quadratic flight path predictor (velocity and acceleration extrapolation)
5)  Data and AI -based enemy ship weak points identifier, for more effective use of weapon
6)  Advanced weapon mounts with predictive cooling request generation (divert more cooling to weapons ahead of overheating)
7)  Weapon Intersperser (fires shield degrading weapons first, hull degrading weapons after, conserving energy and cooling power)
8)  Fleet-wide target sharing system
  Then, a list of rocket and missile weapon support outfits could include,
1)  Burst Launch Control (launch at more than one target simultaneously)
2)  Conservative Launch Control (prevents launching a missile when the likelihood of it maintaining lock is too low, or when the target is already almost dead such that the missile would probably be lost)
3)  Sublime Retargeting (sends new target data to a missile while in flight, when appropriate)
4)  ECCM system that detects ECM of enemy ships and instructs missiles about what false signals to expect and not be fooled by, prior to launching them.
  So, with all these outfits to choose from, would you make a ship that has both beams AND missiles, but that can only fit about half of the outfits of each group?  Or would you design a missile frigate that only has missiles and ALL the missile enhancement outfits, and then a separate beam cruiser design that has only beams and all the outfits that benefit beams?  Obviously the latter option is better.
  And so, by having specialized outfits in great abundance, but limited outfit space, we've introduced the need for the player to think well about what outfits to get or NOT get for any particular ship design.  And the different designs will of course be better at different things, and a fight in space might look a bit more like a complex team-work.  You could even design small ships with no weapons but very agile, that draw fire to themselves but manage to avoid getting hit, to make enemies waste ammo or energy.  You could design a small, weaponless ship to function like an AWACS, with the best sensors and targeting computers, sharing its data with the rest of the fleet by having the Fleet-Wide Target Sharing outfit, thus allowing you to avoid having to install better targeting outfits in all your ships.
  But in this case the AI has to also feature specialized roles that best fit the weapons and outfits of a ship.  That is to say, the AI has to be able to MAKE BEST USE of the weapons and outfits set provided.  Thus, the AWACS ship would stay a safe distance away from danger;  your missile frigates would open battle with a coordinated salvo, then retreat, and let the beamer cruisers take up the mid-fight, then rejoin seconds later with another salvo.  "Bombers" (smaller craft with heavy weapons) would target larger ships in swarms, and interceptors would try to intercept them.
  Do carriers exist in this game?

*  F3:  A chance to reconsider, after turning down a mission.  In the old Privateer game, you could always go back to a character whose mission you turned down, sit at his table again, and say you've reconsidered.  To kill the mission forever is a bit harsh, I find, even if the save file is not encrypted.  But then again, in Privateer everything was visual.  The guy would be sitting at the same table in the bar of the same planet,  even if some other characters showed up at other tables and offered other missions.  Presently, there is only one mission possible through the Spaceport button, and this is the problem.  We could replace the button with a bar and tables, and problem solved.  Or find some other solution, visual or not.

AND THE BIG IDEA:

*  BI1:  Organize a stint of 3D modeling of space-ports:  10 3D modelers working for 2 months to create roughly 40 or 50 bypes of spaceports.  The spaceports can then be rendered in high quality to generate a space-porth image to use as background when you are landed at a planet.  Each model of spaceport should have several doors, which then an image map translates into commands for where to go in or out of the port, such as the shipyard, the market, the outfits, etc.  Each of these places have their own backgrounds, also rendered using the same 3D model, but from inside each of the rooms.

**Comment by Zireael07** [comment]
I think this has too many ideas... They should be split into an issue per idea. (And some of those are I think already on the wishlist, e.g. bigger systems)

**Reply by ChuckStarchaser** [reply]
Okay, I'm going to see if I can resolve the old issue I have had with discord access, as it seems this is not the right place in the first place, I'll have a look at the wishlist, and then split stuff up.  Thanks.

**Comment by 1010todd** [comment]
The big idea of creating 3D render of spaceports is a bit too much at the moment. We have only 3~4 regular modellers all of which are doing it on their free time and often busy with something else. It will very likely take years to model that much with details described and current manpower assuming they're willing in the first place because there's also other things to be made. 

Though there were ideas to split spaceport into multiple sub-location before I don't think each need unique scene yet. Also I think the idea of doors and clicking on it is a bit out of place for Endless Sky style, probably better to just swap other buttons (Trade, Bank, etc.) For another set of buttons upon clicking spaceport.

**Comment by ChuckStarchaser** [comment]
Well, I was suggesting spaceports because I'm a blenderhead, myself, and was trying to give myself a job.
As for the buttons UNDER spaceport, here's the thing:  The whole layout presently does not make sense.  You should not need to click on a button to GO TO the space-port;  you ARE IN the spaceport, when you exit your ship, by definition.  And the Bank, the Trading, the Outfitting and the Shipyard ...  ALL rent spaces inside the space-port.
  With 3D models ...  Well, with images, rendered from 3D models, you simply have a view of the main atrium of the space-port when you land, and a view of the planet through the big glass windows.  In that grand view of the space-port main promenade, there are areas that look like a bank, or like a shipyard, or a bar, etc, and you click on them and you are now in that sub-space (another render).

  But you're right;  this is a lot of work, if fun, even for me.

**Comment by Hecter94** [comment]
Our current artwork bounty list:
### Outfits
* Small and medium reverse ion, plasma, and atomic thrusters. 
* Reverse Wanderer thrusters of all sizes
* A Wanderer scanner intended to be unlocked after the Wanderers go through the Eye.
* Korath Napalm Dispenser: (Requirements list)
    * Korath styling
    * H2H outfit
    * Not a rifle
    * Thing designed to lob globs of flaming Napalm style substances at people.
- Remnant stolen tech outfits: (aka what would they do with all the tech they've stolen from the Korath? 
     * Heat Shunts
     * Plasma cores
     * System cores 
     * Engines
     * Fire-lance 
- Remnant Wanderer tech: (Late-game, the Remnant will contact the Wanderers via the player, particularly during the Wanderer evacuation during and after the Unfettered invasion. What tech might the players help the Remnant get their hands on? What will that tech act and look like when subject to Remnant adaptation and modification?)
- Small, medium, and large salvaging equipment. (Salvage arm, Salvage Suite, Salvage Depot)

### Ships
 - Remnant Loon: small scout/espionage ship (bigger than Penguin, smaller than starling). Long range, moderate speed, cloaking, high-sensors.
 - Remnant Swan: large fuel tanker ship (think big in-flight refueling ship with moderate passenger capacity)
 - Remnant Cormorant: large transport (live-ship / colony ship, carrier)

### Misc.
 - Gas Giant landing screens (for all gas giants in the Ember Waste for now). Versions with and without Void Sprites or other life forms would be appreciated.

**Comment by ChuckStarchaser** [comment]
Thanks!  I'll see if I can tackle some of these ---Remnant--- ... Wanderer requirements.
